Add API docs for Ask CFPB and CCDB5-search

higs4281 opened the pull request:
This is a start. The new pages can be previewed on my fork:
- https://higs4281.github.io/cfgov-refresh/ask-cfpb/
- https://higs4281.github.io/cfgov-refresh/consumer-complaint-database/

This could provide a home for other API references (oah-api?), and maybe for deprecation notes such as mentioned in [GHE]/CFGOV/platform/issues/1001.

cc @stephanieosan @cfarm @amymok

willbarton commented on docs/ask-cfpb.md:
```diff
@@ -0,0 +1,64 @@
+{
+  query: "tuition",
+  result_query: "tuition",
+  results: [
```
After our typo suggestion work, it will also include `suggestion` at the top level of the json object, and if the search term has no results, a suggested term will be there instead. If there is no suggestion, it will be `null`.

https://www.consumerfinance.gov/ask-cfpb/search/json/?q=tuition

higs4281 replied:
Thanks for the suggestion catch! preview is updated:
https://higs4281.github.io/cfgov-refresh/ask-cfpb/

willbarton commented on docs/ask-cfpb.md:
```diff
@@ -0,0 +1,64 @@
+{
+  query: "vehiculo",
+  result_query: "vehiculo",
+  results: [
```
Same as above, `suggestion` will be included.

willbarton commented on docs/consumer-complaint-database.md:
```diff
@@ -0,0 +1,9 @@
+
+The API allows automation of the same filtering and searching functions offered to website visitors.
+
+Detailed documentation for the search API can be found [here](https://higs4281.github.io/ccdb5-api/swagger/index.html).
```
This should live under `cfpb`, not `higs4281` 😉 .

Could you PR that gh-pages branch to [ccdb5-api](https://github.com/cfpb/ccdb5-api/)?

higs4281 replied:
Working on this.

willbarton commented:
Another though, we should really, really change the name of this repo to `consumerfinance.gov` if we're going to use the documentation to document more than just how to work with this particular repo (which is something I absolutely think we should do! I'm pretty I've introduced you to that particular high horse of mine before though :wink:).

higs4281 commented:
Note: The cfgov docs now point to the main ccdb5-api repo instead of my branch. That means the ccdb5 documentation link won't work until a [PR](https://github.com/cfpb/ccdb5-api/pull/69) there is merged.

higs4281 commented:
@willbarton The [CCDB5 changes](https://github.com/cfpb/ccdb5-api/pull/69) have been merged, so this PR should be good to go.

I added also added HMDA API to the lineup.
